### ISSUE TITLE
indexBy output type from input

### ISSUE THREAD

**Issue by gjjones** [HubSpot/transmute#8]
I noticed that `indexBy` determines 2 possible output types that are both `immutable`. I'd think that for an input of JS-`array` that the return out be a JS-`object`.

**Comment by colbyr** [comment]
I totally agree! In fact, I actually did this once but had to revert it because it a broke a bunch of stuff in the CRM https://github.com/HubSpot/transmute/commit/44ef420f165e0185919af1e6b2204f8acebb7904

I can bring it back, but it'll need to be in a new major version.
